x-pack/plugins/encrypted_saved_objects/server/crypto/encrypted_saved_objects_service.ts

```diff
@@ -152,6 +256,46 @@ export class EncryptedSavedObjectsService {
     return this.typeDefinitions.has(type);
   }
 
+  // Embedding excluded AAD fields POC rev2
+  // Initializes versioned type storage/caching
+  // Creates an internal SO repository to write/read the hidden encrpytion
+  // descriptors which contain per-version encrypted fields and AAD metadata
+  public initializeVersionedMetadata(savedObjects: SavedObjectsServiceStart) {
+    this.internalSoRepository = savedObjects.createInternalRepository([
+      VERSIONED_ENCYRPTION_DEFINITION_TYPE,
+    ]);
+
+    const versionedEsoMetadata: SavedObjectsBulkCreateObject[] = [
+      ...this.typeDefinitions.values(),
+    ].flatMap((versionedType) => {
+      return versionedType.modelVersion
+        ? [
+            {
+              id: versionedType.type + versionedType.modelVersion,
+              type: VERSIONED_ENCYRPTION_DEFINITION_TYPE,
+              attributes: {
+                type: versionedType.type,
+                modelVersion: versionedType.modelVersion,
+                attributesToEncrypt: [...versionedType.attributesToEncrypt.values()],
+                attributesToExcludeFromAAD: versionedType.getAttributesToExcludeFromAAD(),
+              },
+            },
+          ]
+        : [];
+    });
+
+    this.versionedMetadataInitialized = this.internalSoRepository
+      .bulkCreate(versionedEsoMetadata, {
+        overwrite: true,
+        managed: true,
+      })
+      .catch((err) => {
```

Comment:
This is not working as I expected it to. We end up with these two errors: https://buildkite.com/elastic/kibana-pull-request/builds/143357#01896a9e-90de-47b3-968d-802656f1725d

The issue appears to be the `.catch(...)`, and I could use some help understanding why.